core/src/main/java/org/apache/spark/util/collection/unsafe/sort/RadixSort.java

```diff
@@ -40,28 +40,28 @@
    *         of always copying the data back to position zero for efficiency.
    */
   public static int sort(
-      LongArray array, int numRecords, int startByteIndex, int endByteIndex,
+      LongArray array, long numRecords, long startByteIndex, long endByteIndex,
```

Comment:
I think we should keep the byte indices as ints, since they are in 0..7 and we assert these below.